http/src/main/java/com/github/awsjavakit/http/OAuth2HttpClient.java

```diff
@@ -4,78 +4,169 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.awsjavakit.misc.JacocoGenerated;
 import com.github.awsjavakit.misc.paths.UriWrapper;
 import java.io.IOException;
-import java.net.URI;
+import java.net.Authenticator;
+import java.net.CookieHandler;
+import java.net.ProxySelector;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
 import java.net.http.HttpResponse.BodyHandler;
 import java.net.http.HttpResponse.BodyHandlers;
+import java.net.http.HttpResponse.PushPromiseHandler;
 import java.nio.charset.StandardCharsets;
+import java.time.Duration;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.BiPredicate;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+
+/**
+ * A wrapper of HttpClient that performs an OAuth2 authentication of "grant_type"
```

Comment:
This time the client has been implemented as a proper decorator, i.e., it extends the contained class. This gives us more flexibility because now the OAuth2HttpClient IS an HttpClient

Reply:
Very nice